### PR TITLE
reverseproxy: Configurable forward proxy URL

### PR DESCRIPTION
**Pull request by ImpostorKeanu**
Closes: #6111 

Apologies for the redo. I'm security person by trade so I often fumble through git.

This PR should incorporate corrections discussed in the last PR that I closed.

Grateful for the patience.

### CODE REVIEW

File: modules/caddyhttp/reverseproxy/httptransport.go [hunk 1]
```diff
@@ -71,6 +72,22 @@ type HTTPTransport struct {
...
+	//
+	// Providing a value to this parameter results in
+	// requests flowing through the reverse_proxy in the following
+	// way:
+	//
+	// User Agent ->
+	//  reverse_proxy ->
+	//  forward_proxy_url -> upstream
+	//
+	// Default: off.
```
Comment (WeidiDeng) [inline modules/caddyhttp/reverseproxy/httptransport.go:88]: Default is not `off`. It's read from environmental variables.

File: modules/caddyhttp/reverseproxy/httptransport.go [hunk 2]
```diff
@@ -265,8 +282,23 @@ func (h *HTTPTransport) NewTransport(caddyCtx caddy.Context) (*http.Transport, e
...
 
+	// negotiate any HTTP/SOCKS proxy for the HTTP transport
+	var proxy func(*http.Request) (*url.URL, error)
+	if h.ForwardProxyURL != "" {
+		pUrl, err := url.Parse(h.ForwardProxyURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse transport proxy url: %v", err)
+		}
+		caddyCtx.Logger().Info("setting transport proxy url", zap.String("url", h.ForwardProxyURL))
+		proxy = func(request *http.Request) (*url.URL, error) {
```
Comment (WeidiDeng) [inline modules/caddyhttp/reverseproxy/httptransport.go:293]: Should just use [`http.ProxyURL`](https://pkg.go.dev/net/http#ProxyURL).

**Review by WeidiDeng — CHANGES_REQUESTED** [review]

File: modules/caddyhttp/reverseproxy/httptransport.go (hunk 1 above)
Comment (ImpostorKeanu) [reply modules/caddyhttp/reverseproxy/httptransport.go:88]: The intent is for the individual module to be able to be configured with a specific proxy. `ForwardProxyURL` takes precedence over `HTTP_PROXY` and no default value is supplied.

Comment (francislavoie) [reply modules/caddyhttp/reverseproxy/httptransport.go:88]: But `if h.ForwardProxyURL == ""` it still uses `http.ProxyFromEnvironment` therefore the default is still "from environment", not nothing.

**Review by mholt — APPROVED** [review]
Thanks for the contribution! Is this ready to merge? It LGTM.

**Review by WeidiDeng — COMMENTED** [review]
Maybe Caddyfile support should be added? For now it's only configurable through JSON, which is very tedious.

**Review by WeidiDeng — APPROVED** [review]

**Pull request merged**